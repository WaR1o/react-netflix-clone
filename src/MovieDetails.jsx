import React from "react";
import moment from "moment";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { Poster, DetailedMovie, Year, HeaderWrapper } from "./styles";
import { getMovie, resetMovie } from "./store/actions";

class MovieDetails extends React.Component {
  componentDidMount() {
    const { getMovie, match } = this.props;
    getMovie(match.params.id);
  }

  componentWillReceiveProps(nextProps) {
    if (!this.props.location || !nextProps.location) {
      return;
    }
    if (this.props.location.pathname !== nextProps.location.pathname) {
      this.props.getMovie(nextProps.match.params.id);
    }
  }

  componentWillUnmount() {
    const { resetMovie } = this.props;
    resetMovie();
  }

  render() {
    const { movie } = this.props;
    return (
      <HeaderWrapper>
        <DetailedMovie>
          <Poster src={movie.poster_path} alt="movie poster" />
          <div>
            <h2>{movie.title}</h2>
            <span>{movie.vote_average}</span>
            <br />
            <span>{moment(movie.release_date).format("YYYY")}</span>
            <br />
            <span>{movie.runtime} min</span>
            <p>{movie.overview}</p>
          </div>
        </DetailedMovie>
      </HeaderWrapper>
    );
  }
}

const mapStateToProps = state => ({
  movies: state.movies,
  movie: state.movie
});

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      getMovie,
      resetMovie
    },
    dispatch
  );

export default connect(mapStateToProps, mapDispatchToProps)(MovieDetails);
