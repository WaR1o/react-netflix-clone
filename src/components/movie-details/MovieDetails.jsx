import React from 'react';
import moment from 'moment';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { Poster, Year, HeaderWrapper } from '../../styles/common';
import { DetailedMovie } from './MovieDetailsStyles';
import { getMovie, resetMovie } from '../../store/actions';

class MovieDetails extends React.Component {
  componentDidMount() {
    const { getMovie, match } = this.props;
    if (getMovie && match) {
      getMovie(match.params.id);
    }
  }

  componentWillReceiveProps(nextProps) {
    if (!this.props.location || !nextProps.location) {
      return;
    }
    if (this.props.location.pathname !== nextProps.location.pathname) {
      this.props.getMovie(nextProps.match.params.id);
    }
  }

  componentWillUnmount() {
    const { resetMovie } = this.props;
    resetMovie();
  }

  render() {
    const { movie } = this.props;
    return (
      <HeaderWrapper>
        <DetailedMovie>
          <Poster src={movie.poster_path} alt="movie poster" />
          <div>
            <h2>{movie.title}</h2>
            <span>{movie.vote_average}</span>
            <br />
            <span>{moment(movie.release_date).format('YYYY')}</span>
            <br />
            <span>{movie.runtime} min</span>
            <p>{movie.overview}</p>
          </div>
        </DetailedMovie>
      </HeaderWrapper>
    );
  }
}

const mapStateToProps = state => ({
  movies: state.movies,
  movie: state.movie
});

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      getMovie,
      resetMovie
    },
    dispatch
  );

const ConnectedMovieDetails = connect(mapStateToProps, mapDispatchToProps)(MovieDetails);
export const UnwrappedMovieDetails = MovieDetails;
export { ConnectedMovieDetails as MovieDetails };
