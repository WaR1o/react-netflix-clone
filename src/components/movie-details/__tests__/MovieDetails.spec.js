/* eslint-env node, jest */
import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { UnwrappedMovieDetails } from '../MovieDetails.jsx';
import { SINGLE_MOVIE } from '../../../mock-data';

configure({ adapter: new Adapter() });

test('MovieDetails renders correctly ', () => {
  const movie = SINGLE_MOVIE;
  const component = shallow(<UnwrappedMovieDetails movie={movie} />);
  expect(component).toMatchSnapshot();
});
