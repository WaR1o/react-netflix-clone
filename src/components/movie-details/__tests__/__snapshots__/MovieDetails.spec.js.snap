// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails renders correctly  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieDetails
    movie={
      Object {
        "budget": 55000000,
        "genres": Array [
          "Drama",
          "Romance",
        ],
        "id": 337167,
        "overview": "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
        "poster_path": "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
        "release_date": "2018-02-07",
        "revenue": 136906000,
        "runtime": 106,
        "tagline": "Don't miss the climax",
        "title": "Fifty Shades Freed",
        "vote_average": 6.1,
        "vote_count": 1195,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styled.div>
        <styled.img
          alt="movie poster"
          src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
        />
        <div>
          <h2>
            Fifty Shades Freed
          </h2>
          <span>
            6.1
          </span>
          <br />
          <span>
            2018
          </span>
          <br />
          <span>
            106
             min
          </span>
          <p>
            Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.
          </p>
        </div>
      </styled.div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.img
            alt="movie poster"
            src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
          />,
          <div>
            <h2>
              Fifty Shades Freed
            </h2>
            <span>
              6.1
            </span>
            <br />
            <span>
              2018
            </span>
            <br />
            <span>
              106
               min
            </span>
            <p>
              Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.
            </p>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alt": "movie poster",
            "src": "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Fifty Shades Freed
              </h2>,
              <span>
                6.1
              </span>,
              <br />,
              <span>
                2018
              </span>,
              <br />,
              <span>
                106
                 min
              </span>,
              <p>
                Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.
              </p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Fifty Shades Freed",
              },
              "ref": null,
              "rendered": "Fifty Shades Freed",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 6.1,
              },
              "ref": null,
              "rendered": 6.1,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2018",
              },
              "ref": null,
              "rendered": "2018",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  106,
                  " min",
                ],
              },
              "ref": null,
              "rendered": Array [
                106,
                " min",
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
              },
              "ref": null,
              "rendered": "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div>
          <styled.img
            alt="movie poster"
            src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
          />
          <div>
            <h2>
              Fifty Shades Freed
            </h2>
            <span>
              6.1
            </span>
            <br />
            <span>
              2018
            </span>
            <br />
            <span>
              106
               min
            </span>
            <p>
              Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.
            </p>
          </div>
        </styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.img
              alt="movie poster"
              src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
            />,
            <div>
              <h2>
                Fifty Shades Freed
              </h2>
              <span>
                6.1
              </span>
              <br />
              <span>
                2018
              </span>
              <br />
              <span>
                106
                 min
              </span>
              <p>
                Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.
              </p>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "movie poster",
              "src": "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  Fifty Shades Freed
                </h2>,
                <span>
                  6.1
                </span>,
                <br />,
                <span>
                  2018
                </span>,
                <br />,
                <span>
                  106
                   min
                </span>,
                <p>
                  Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.
                </p>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Fifty Shades Freed",
                },
                "ref": null,
                "rendered": "Fifty Shades Freed",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 6.1,
                },
                "ref": null,
                "rendered": 6.1,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "2018",
                },
                "ref": null,
                "rendered": "2018",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    106,
                    " min",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  106,
                  " min",
                ],
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
                },
                "ref": null,
                "rendered": "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
